webapp/src/main/webapp/vue-apps/Admin/components/AdminApp.vue

```diff
@@ -102,27 +102,35 @@ export default {
     return {
       providers: [],
       switcher: false,
-      error: null
+      error: null,
+      log: null
     };
   },
   created() {
+    this.log = webConferencing.getLog().prefix("webconferencing.admin");
     this.getProviders();
   },
   methods: {
     async getProviders() {
       // services object contains urls for requests
       try {
-        // const data = await getData(this.services.providers);
         const response = await webConferencing.getProvidersConfig();
         const data = webConferencing.getProvidersConfig().done(response);
         this.error = null;
         this.providers = await response;
-        // const resourcesPromises = this.providers.map(({ provider }) => this.getProviderResources(provider));
-        // Promise.all(resourcesPromises).then(res => {
-        //   res.map(localized => {
-        //     this.i18n.mergeLocaleMessage(this.language, localized.getLocaleMessage(this.language));
-        //   });
-        // });
+
+        for (const p of this.providers) {
+          webConferencing.getProvider(p.type).done(function(provider) {
+            p.val = provider;
+            if (provider.showSettings && provider.hasOwnProperty("showSettings")) {
```

Comment:
This could be improved:
`p.hasSettings = !(provider.showSettings && provider.hasOwnProperty("showSettings"));`

Reply:
Yes. This's improved